### PR TITLE
Changed TaskApplicationDescriptor APIs to be fluent and consistent with StreamApplicationDescriptor API.

### PR DESCRIPTION
**Pull request by prateekm**

### CODE REVIEW

**Comment by prateekm** [comment]
@nickpan47 @rmatharu @bharathkk Please take a look.

File: samza-api/src/main/java/org/apache/samza/application/TaskApplication.java [hunk 1]
```diff
@@ -50,9 +50,10 @@
  *     KafkaOutputDescriptor<PageViewEvent>> outputStreamDescriptor =
  *         trackingSystemDescriptor.getOutputDescriptor("recentPageViewEvent", new JsonSerdeV2<>(PageViewEvent.class)));
  *
- *     appDescriptor.addInputStream(inputStreamDescriptor);
- *     appDescriptor.addOutputStream(outputStreamDescriptor);
- *     appDescriptor.setTaskFactory((StreamTaskFactory) () -> new PageViewTask());
+ *     appDescriptor
+ *         .addInputStream(inputStreamDescriptor)
+ *         .addOutputStream(outputStreamDescriptor)
+ *         .withTaskFactory((StreamTaskFactory) () -> new PageViewTask());
```
Comment (rmatharu) [inline samza-api/src/main/java/org/apache/samza/application/TaskApplication.java]: Doesn't setTaskFactory seem more consistent with addInput and addOutput, 
vs. withTaskFactory consistent with withInput withOutput?

Comment (prateekm) [reply samza-api/src/main/java/org/apache/samza/application/TaskApplication.java]: Other methods in the base ApplicationDescriptor are already called withXYZ. We use the same convention in I/O and Table descriptors for setters as well, so prefer consistency.

I do think the `withTaskFactory, withInputStream and withOutputStream` set of methods makes sense. @nickpan47 WDYT?

Comment (nickpan47) [reply samza-api/src/main/java/org/apache/samza/application/TaskApplication.java]: +1. Sounds good to me.

File: samza-api/src/main/java/org/apache/samza/application/TaskApplication.java [hunk 2]
```diff
@@ -50,9 +50,10 @@
  *     KafkaOutputDescriptor<PageViewEvent>> outputStreamDescriptor =
  *         trackingSystemDescriptor.getOutputDescriptor("recentPageViewEvent", new JsonSerdeV2<>(PageViewEvent.class)));
  *
- *     appDescriptor.addInputStream(inputStreamDescriptor);
- *     appDescriptor.addOutputStream(outputStreamDescriptor);
- *     appDescriptor.setTaskFactory((StreamTaskFactory) () -> new PageViewTask());
+ *     appDescriptor
+ *         .addInputStream(inputStreamDescriptor)
```
Comment (nickpan47) [inline samza-api/src/main/java/org/apache/samza/application/TaskApplication.java]: Can we change the comments here to withXXX().withXXX()?

Comment (prateekm) [reply samza-api/src/main/java/org/apache/samza/application/TaskApplication.java]: 👍 Will change them ti withInput/OutputStream.

**Pull request closed without merging**